pandas/io/formats/csvs.py

```diff
@@ -252,22 +255,58 @@ def save(self) -> None:
             storage_options=self.storage_options,
         ) as handles:
             # Note: self.encoding is irrelevant here
+            self._save(handles.handle)
+
+    def _save_pyarrow(self, handle) -> None:
+        pa = import_optional_dependency("pyarrow")
+        pa_csv = import_optional_dependency("pyarrow.csv")
+        # Convert index to column and rename name to empty string
+        # since we serialize the index as basically a column with no name
+        # TODO: this won't work for multi-indexes
+        obj = self.obj.reset_index(names=[""])
```

Comment:
Can we explicitly set the name to `None` after the reset_index?

Reply:
We don't want None, but actually this empty string (for compatibility with current to_csv). 

But, we should only rename to `""` if the index has no name itself, i.e. only when the name was originally `None`. 

So I think we can do something like:

```suggestion
        new_names = [label if label is not None else "" for label in self.obj.index.names]
        obj = self.obj.reset_index(names=new_names)
```

and then that should also work fine for MultiIndex?

Reply:
Thanks for the modifications. I tested it and it seems to work (just a small issue with pyarrow quoting all strings).

What if the MultiIndex doesn't have names, though?
IIRC, pyarrow doesn't allow duplicate column names, so the trick wouldn't work anymore (can't have two ``""`` columns).

Reply:
Hello @lithomas1 , how to make sure that pyarrow and pandas have the same quoting results? In pyarrow 12.0.1, setting `quoting_style` to `needed`(the default value) will quote all strings, but by default pandas will only quote string if needed.